tests/provider_test.py

```diff
@@ -0,0 +1,120 @@
+''' Test to check providers '''
+import unittest
+
+from helpers.providers.http_provider import (GenericHTTPProvider,
+                                             GitHubHTTPProvider,
+                                             GitlabHTTPProvider)
+
+# pylint: disable=protected-access
+
+
+class GitHubHTTPProviderTests(unittest.TestCase):
+    ''' GitHub Provider Tests '''
+
+    def setUp(self):
+        self.provider = GitHubHTTPProvider
+        self.test_url = 'https://github.com/Bonial-International-GmbH/MkRadar/blob/main/README.md'
+
+    def test_can_open(self):
+        ''' Tests if provider check correctly if it can handle the url '''
+        self.assertTrue(self.provider.can_open(self.test_url))
+        self.assertFalse(self.provider.can_open(
+            'github.com:Bonial-International-GmbH/MkRadar.git'))
+        self.assertFalse(self.provider.can_open(
+            'https://bitbucket.org/MysteryInc/MagicProject/src/master/'))
+
+    def test_pre_url_modify(self):
+        ''' Tests if the url are correctly re-writen '''
+        self.assertEqual(
+            self.provider._pre_url_modify(self.test_url),
+            'https://raw.githubusercontent.com/Bonial-International-GmbH/MkRadar/main/README.md'
+        )
+
+    def test_unauthenticated(self):
+        ''' Test if provider can download unauthenticated '''
+        code = self.provider._get_page_unauthorized(self.test_url).status_code
+        self.assertEqual(200, code)
```

Comment:
As far as I know unit tests should be isolated, here you are assuming during the running test phase you will have internet access which for me belongs to integrations tests

Reply:
True, but I don't know how to mook this.